src/main/scala/com/vngrs/etl/transformers/ToCsvTransformer.scala

```diff
@@ -0,0 +1,86 @@
+package com.vngrs.etl.transformers
+
+import java.text.SimpleDateFormat
+
+import com.vngrs.etl.Transform
+import com.vngrs.etl.codecs.ToCsvCodec
+import com.vngrs.etl.configs.{ToCsvCodecConfigs, ToCsvTransformerConfigs}
+import org.apache.commons.csv.CSVFormat
+import org.apache.spark.rdd.RDD
+
+/**
+  * Transforms [[A]]s to `CSV` formatted [[String]]s with help of given `toCsvCodec`.
+  *
+  * @param toCsvCodec [[A]] to `CSV` codec
+  * @param config To `CSV` transformer configurations
+  * @tparam A Type of the data ([[org.apache.spark.rdd.RDD]]).
+  */
+@SuppressWarnings(Array("org.wartremover.warts.Any", "org.wartremover.warts.AsInstanceOf", "org.wartremover.warts.Var"))
+abstract class ToCsvTransformer[A]
+(
+  toCsvCodec: ToCsvCodec[A],
+  config: ToCsvTransformerConfigs
+) extends Transform[A, String] with Serializable {
+
+  /**
+    * Transforms [[A]]s to `CSV` formatted [[String]]s with help of given `toCsvCodec`.
+    *
+    * @param input [[org.apache.spark.rdd.RDD]]
+    * @return Transformed [[org.apache.spark.rdd.RDD]]
+    */
+  override def apply(input: RDD[A]): RDD[String] = {
+
+    val header =
+      if (config.generateHeader) {
+        createHeader(input)
+      } else {
+        Array[AnyRef]()
+      }
+
+    input.mapPartitions[String](iterator =>
```

Comment:
Can we implement simpler? Looks like it only 2 things:
1- Convert elements to CSV string
2- Add headers if configured to generate headers

So it could be implemented like
```scala
val mapped = xs.map(x => csvFormat.format(toCsvCodec(c, codecConf): _*))
if (config.generateHeader) csvFormat.format(header: _*) :: mapped else mapped
```